pkg/virt-launcher/virtwrap/network/podnic_test.go

```diff
@@ -0,0 +1,149 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"runtime"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/golang/mock/gomock"
+	"github.com/vishvananda/netlink"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	kubevirtv1 "kubevirt.io/client-go/api/v1"
+	v1 "kubevirt.io/client-go/api/v1"
+	"kubevirt.io/kubevirt/pkg/network/cache"
+	"kubevirt.io/kubevirt/pkg/network/cache/fake"
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+	neterrors "kubevirt.io/kubevirt/pkg/network/errors"
+	"kubevirt.io/kubevirt/pkg/network/infraconfigurators"
+	"kubevirt.io/kubevirt/pkg/virt-launcher/virtwrap/api"
+)
+
+var _ = Describe("podNIC", func() {
+	var (
+		mockNetwork                *netdriver.MockNetworkHandler
+		cacheFactory               cache.InterfaceCacheFactory
+		mockPodNetworkConfigurator *infraconfigurators.MockPodNetworkInfraConfigurator
+		ctrl                       *gomock.Controller
+		vmi                        *kubevirtv1.VirtualMachineInstance
+	)
+	newPodNICWithMocks := func(vmi *v1.VirtualMachineInstance) *podNIC {
+		podnic, err := newPodNICWithWithoutInfraConfigurator(vmi, &vmi.Spec.Networks[0], mockNetwork, cacheFactory, nil)
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
```

Comment:
I still think this is the worst option of the choices you had, but won't block on it.

Reply:
Ok, let me revisit it returning the error, it does not add too much verbosity to the tests.

Reply:
It's OK, my bad on giving you choices.